apps/prairielearn/src/question-servers/index.ts

```diff
@@ -1,115 +1,6 @@
-}
-
-export type PrepareResultData = GenerateResultData;
-
-export interface RenderResultData {
-  extraHeadersHtml: string;
-  questionHtml: string;
-  submissionHtmls: string[];
-  answerHtml: string;
-}
-
-export interface ParseResultData {
-  params: Record<string, any>;
-  true_answer: Record<string, any>;
-  submitted_answer: Record<string, any>;
-  feedback: Record<string, any>;
-  raw_submitted_answer: Record<string, any>;
-  format_errors: Record<string, any>;
-  gradable: boolean;
-}
-
-export interface GradeResultData {
-  params: Record<string, any>;
-  true_answer: Record<string, any>;
-  submitted_answer: Record<string, any>;
-  format_errors: Record<string, any>;
-  raw_submitted_answer: Record<string, any>;
-  partial_scores: Record<string, any>;
-  score: number;
-  feedback: Record<string, any>;
-  gradable: boolean;
-  v2_score?: number;
-}
-
-export interface TestResultData {
-  params: Record<string, any>;
-  true_answer: Record<string, any>;
-  format_errors: Record<string, any>;
-  raw_submitted_answer: Record<string, any>;
-  partial_scores: Record<string, any>;
-  score: number;
-  gradable: boolean;
-}
-
-export interface QuestionServer {
-  generate: (
-    question: Question,
-    course: Course,
-    variant_seed: string,
-  ) => QuestionServerReturnValue<Partial<GenerateResultData>>;
-  prepare: (
-    question: Question,
-    course: Course,
-    variant: {
-      variant_seed: string;
-      params: Record<string, any>;
-      true_answer: Record<string, any>;
-      options: Record<string, any>;
-      broken: boolean;
-    },
-  ) => QuestionServerReturnValue<PrepareResultData>;
-  render: (
-    renderSelection: { question: boolean; answer: boolean; submissions: boolean },
-    variant: Variant,
-    question: Question,
-    submission: Submission | null,
-    submissions: Submission[],
-    course: Course,
-    locals: Record<string, any>,
-  ) => QuestionServerReturnValue<RenderResultData>;
-  parse: (
-    submission: Pick<
-      Partial<Submission>,
-      'submitted_answer' | 'feedback' | 'format_errors' | 'raw_submitted_answer' | 'gradable'
-    >,
-    variant: Variant,
-    question: Question,
-    course: Course,
-  ) => QuestionServerReturnValue<ParseResultData>;
-  grade: (
-    submission: Submission,
-    variant: Variant,
-    question: Question,
-    course: Course,
-  ) => QuestionServerReturnValue<Partial<GradeResultData>>;
-  file?: (
-    filename: string,
-    variant: Variant,
-    question: Question,
-    course: Course,
-  ) => QuestionServerReturnValue<Buffer>;
-  test?: (
-    variant: Variant,
-    question: Question,
-    course: Course,
-    test_type: 'correct' | 'incorrect' | 'invalid',
-  ) => QuestionServerReturnValue<TestResultData>;
-}
+import { type EffectiveQuestionType, type QuestionServer, type QuestionType } from './types.js';
+// Re-export all the types from types.js
+export * from './types.js';
 
 const questionModules: Record<EffectiveQuestionType, QuestionServer> = {
```

Comment:
Nit, we can write this as `... = { ... } satisfies Record<EffectiveQuestionType, QuestionServer>` for slightly better type checking: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator

Reply:
Resolved in e43a9176c261d75423686e996750b76aea81674d.